leapp/cli/upgrade/__init__.py

```diff
@@ -211,6 +212,12 @@ def preupgrade(args):
         report_info(get_config().get('report', 'path'), fail=True)
         sys.exit(1)
     report_info(get_config().get('report', 'path'))
+    # fetch all report messages as a list of dicts
+    messages = fetch_upgrade_report_raw(context, renderers=False)
+    report_json = get_config().get('report', 'path').replace('.txt', '.json')
+    # XXX FIXME use leapp way of outputting things?
+    with open(report_json, 'w+') as f:
+        f.write(json.dumps({'entries': messages}))
```

Comment:
json.dump? I don’t see why you need to create a temporary string - I know WIP and POC etc 
Just saying:-)